low_level_platform/impl/src/lf_STM32f4_support.c

```diff
@@ -0,0 +1,238 @@
+#if defined(PLATFORM_STM32F4)
+
+#include "platform/lf_STM32f4_support.h"
+#include "low_level_platform.h"
+#include "tag.h"
+
+//  + -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- +
+//  | Important defines and global variables
+//  + -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- +
+
+static volatile bool _lf_sleep_interrupted = false;
+static volatile bool _lf_async_event = false;
+
+// nested critical section counter
+static uint32_t _lf_num_nested_crit_sec = 0;
+
+// Timer upper half (for overflow)
+static uint32_t _lf_time_us_high = 0;
+
+#define LF_MIN_SLEEP_NS 10
+
+// Combine 2 32bit works to a 64 bit word (Takes from nrf52 support)
+#define COMBINE_HI_LO(hi, lo) ((((uint64_t)hi) << 32) | ((uint64_t)lo))
+
+void lf_SystemClock_Config();
+void Error_Handler();
+
+//  + -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- +
+//  | Code for timer functions
+//  + -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- +
+// We use timer 5 for our clock (probably better than dealing with sysTick)
+void _lf_initialize_clock(void) {
+  // Standard initializations from generated code
+  HAL_Init();
+  lf_SystemClock_Config();
+
+  // Configure TIM5 as our 32-bit clock timer
+  __HAL_RCC_TIM5_CLK_ENABLE(); // initialize counter
+  TIM5->CR1 = TIM_CR1_CEN;     // enable counter
+
+  // set prescaler to (16 - 1) = 15
+  // CPU runs a 16MHz so timer ticks at 1MHz
+  //      Which means period of 1 microsecond
+  TIM5->PSC = 15;
+
+  // Setup ISR to increment upper bits
+  TIM5->DIER |= TIM_DIER_CC1IE;
+  NVIC_EnableIRQ(TIM5_IRQn);
+
+  /* This is to make the Prescaler actually work
```

Comment:
I tried playing around with the order but it seems like it still didnt work. So unless I did something wrong with my implementation, I dont know if there is a way to avoid this